src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -186,10 +186,22 @@ private PrimaryAffiliationEvent createPrimaryAffiliationEvent(UserEvent userEven
     User userDto = userEvent.getUserDto();
     if (StringUtils.isNotBlank(userDto.getUsername())) { // for delete event username will be empty
       event.setUsername(userEvent.getUserDto().getUsername());
-      if (ObjectUtils.isNotEmpty(userDto.getPersonal())) {
-        event.setEmail(userDto.getPersonal().getEmail());
-        event.setPhoneNumber(userDto.getPersonal().getPhone());
-        event.setMobilePhoneNumber(userDto.getPersonal().getMobilePhone());
+
+      var personalInfo = userDto.getPersonal();
+      if (ObjectUtils.isNotEmpty(personalInfo)) {
+        event.setEmail(personalInfo.getEmail());
+        event.setPhoneNumber(personalInfo.getPhone());
+        event.setMobilePhoneNumber(personalInfo.getMobilePhone());
+      }
+
+      String barcode = userDto.getBarcode();
+      if (ObjectUtils.isNotEmpty(barcode)) {
```

Comment:
You can avoid these checks here, NPE will not be thown here even if brcode is null,
so  please use :
event.setBarcode(barcode);

The same for externalSystemId

Reply:
fixed